issue-cache-collection.md

```diff
@@ -0,0 +1,148 @@
+5) Time source: Use monotonic clock for scheduling and system clock for timestamps. Confirm precision and any clock sync requirements.
+
+Acceptance Criteria
+- `resctrl` crate exposes an API to read LLC occupancy for a given monitor group, with unit tests for path parsing and domain enumeration.
+- `resctrl-collector` crate:
+  - Initializes/consumes NRI resctrl and metadata signals in-process.
+  - Maintains a mapping from identifiers to monitor groups and metadata.
+  - Samples all known groups at 1 Hz by default and emits Arrow RecordBatches over a channel.
+  - Handles group removal and transient read failures gracefully.
+- Collector integrates the RecordBatch channel into the existing Parquet writer, producing Parquet files in the configured storage.
+- Feature is gated by config and disabled by default.
+
+Implementation Plan (high-level)
+- resctrl crate
+  - [ ] Add domain discovery under a monitor group (e.g., `mon_data/mon_L3_*`).
+  - [ ] Add `llc_occupancy_bytes` read API returning per-domain values and a convenience aggregator.
+  - [ ] Unit tests using a fake filesystem layout to validate parsing and error cases.
+- resctrl-collector crate
+  - [ ] Define public API: builder/config, `run()` task, output `mpsc::Receiver<RecordBatch>`.
+  - [ ] Wire NRI resctrl plugin events to internal state (add/update/remove monitor groups, include group path and association keys).
+  - [ ] Wire NRI metadata plugin events to internal label state.
+  - [ ] Implement 1 Hz sampling loop (Tokio interval), reading via `resctrl` crate and building RecordBatches.
+  - [ ] Bounded channel and backpressure strategy; metrics/logging on drops.
+  - [ ] Basic integration tests behind a feature flag with a mocked `resctrl` trait.
+- Collector integration
+  - [ ] Config surface `resctrl_collector` with defaults.
+  - [ ] Instantiate `resctrl-collector` when enabled and connect its channel to the existing Parquet writer.
+  - [ ] Define Parquet schema and ensure partitioning/rotation align with existing conventions.
+
+Risks & Mitigations
+- Hardware/kernel dependency: resctrl not present or disabled. Mitigate with feature gating and clear diagnostics; no-op when unavailable.
+- Permissions: insufficient privileges to read resctrl. Document required capabilities and provide startup checks.
+- Overhead: sampling loop should be light, but ensure minimal file reads and reuse opened descriptors where possible.
+- Label churn: late/changed metadata could produce cardinality issues. Use stable keys (UIDs) and update labels carefully.
+
+Notes on What’s Structurally Required (must-have to work)
+- NRI resctrl plugin must create monitor groups and keep container/pod tasks assigned to those groups; otherwise data is meaningless.
+- The plugin must expose identifiers sufficient to map monitor group paths to pods/containers (e.g., pod UID, container ID) via an in-process API or well-defined IPC that `resctrl-collector` consumes.
```

Comment:
it is an in-process API via mpsc as discussed above (you can examine `lib.rs` in `nri-resctrl-plugin`)